dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/IdentifierPatternToFileResourceUriResolverImpl.java

```diff
@@ -130,7 +132,10 @@ public List<URI> getUris(String identifier) throws ResourceIOException {
   public List<URI> getUris(String identifier, MimeType mimeType) throws ResourceIOException {
     final List<URI> uris = getUris(identifier);
     return uris.stream()
-        .filter(u -> (mimeType.matches(MimeType.fromURI(u)) || MimeType.fromURI(u) == null))
+        .filter(
+            u ->
+                (mimeType.matches(MimeType.fromURI(u))
+                    || MimeType.fromURI(u) == MIME_APPLICATION_OCTET_STREAM))
```

Comment:
Besser `.equals` statt `==`. In Java vergleicht `==` die Identität, und das ist in diesem Fall vermutlich kein Problem weil das im Hintergrund auf dieselbe Konstante verweist oder der String interned ist, aber prinzipiell safe ist es nicht.
```suggestion
                    || MimeType.fromURI(u).equals(MIME_APPLICATION_OCTET_STREAM)))
```